crates/sui-core/src/quorum_driver/mod.rs

```diff
@@ -757,7 +758,14 @@ where
         mut task_receiver: Receiver<QuorumDriverTask>,
         metrics: Arc<QuorumDriverMetrics>,
     ) {
+        let limit = Arc::new(Semaphore::new(TASK_QUEUE_SIZE));
         while let Some(task) = task_receiver.recv().await {
+            let limit = limit.clone();
```

Comment:
why is this needed while we clone again in line 766?

Reply:
Removed.